react/week2/todo/src/Todo.js

```diff
@@ -0,0 +1,81 @@
+
+import React, { useState } from 'react';
+import Item from './Item';
+import TodoForm from "./TodoForm";
+
+const Todo = () => {
+
+
+	const taskItem = [
+		{
+			text: 'Get out of bed',
+			isCompleted: false
+		},
+
+		{
+			text: 'Brush teeth',
+			isCompleted: false
+		},
+
+		{
+			text: 'Eat breakfast',
+			isCompleted: false
+		},
+	];
+
+	/**
+	 * todos: is taskItem, whose value will be equal to taskItem that we pass in useState() as a parameter
+	 * setTodos is like setState
+	 */
+	const [ todos, setTodos ] = useState( taskItem );
+
+	const addToDo = ( text ) => {
+		const newToDos = [ ...todos, { text } ];
+		setTodos( newToDos );
+	};
+
+	const handleItemClick = ( index ) => {
+		// Get all todos array from state.
+		const newTodos = [ ...todos ];
+
+		// Set isCompleted property to reverse of what its current value is ( boolean )
+		newTodos[ index ].isCompleted = ! newTodos[ index ].isCompleted;
```

Comment:
Even though it might seem, as you are modifying the item inside of a new (copied) array, `newTodos[index]` still references the object from the original `todos` array. You can see it by logging out the item before and after the line 42:

```
    console.log(todos[index])
    // Set isCompleted property to reverse of what its current value is ( boolean )
    newTodos[index].isCompleted = !newTodos[index].isCompleted;
    console.log(todos[index])
```

Hence you are changing the object from the state directly, which is not recommended. Please check this comment I left on a similar issue to get an insight how to fix it:

https://github.com/cecastosic/hyf-homework/pull/12#discussion_r376589618